deepinv/models/latentden.py

```diff
@@ -0,0 +1,171 @@
+import torch
+import torch.nn as nn
+
+
+class LatentDiffusion(nn.Module):
+    r"""
+    Stable-Diffusion v1.5 latent model wrapper (UNet + VAE + CLIP).
+
+    This module loads SD-1.5 components directly from the Hugging Face repo
+    (default: ``runwayml/stable-diffusion-v1-5``) and exposes:
+
+    * :py:meth:`forward` — predicts the noise :math:`\epsilon_\theta(x_t, t)` for a latent
+      :math:`x_t` and timestep ``t`` using Classifier-Free Guidance (CFG).
+    * :py:meth:`encode` — VAE encode image :math:`\to` latent (applies SD scaling factor).
+    * :py:meth:`decode` — VAE decode latent :math:`\to` image (applies SD scaling factor).
+
+    .. note::
+       No scheduler/pipeline is created here; your sampler (e.g., DDIM/PSLD) must supply timesteps.
+       The VAE interface here works in the range ``[-1, 1]`` for images.
+
+    :param int num_inference_steps: kept for API symmetry; not used internally.
+    :param float guidance_scale: CFG scale (``1.0`` effectively disables CFG).
+    :param str prompt: kept for API symmetry; not used internally.
+    :param int height: kept for API symmetry; not used internally.
+    :param int width: kept for API symmetry; not used internally.
+    :param str | torch.device device: device to load models on (e.g., ``"cuda"``).
+    :param torch.dtype dtype: weights/activations dtype for UNet/VAE/TextEncoder (default: ``torch.float16``).
+    :param str model_id: Hugging Face repo id for SD-1.5 components.
+    """
+
+    def __init__(
+        self,
+        guidance_scale=7.5,  # CFG scale
+        device="cuda",
```

Comment:
I suggest make the default device to CPU as otherwise it fails on a CPU-only system
```suggestion
        device: str | torch.device ="cpu",
```